keychain_mac.cpp

```diff
@@ -203,13 +212,19 @@ void deletePassword(const std::string &package, const std::string &service,
         NULL, // unused output parameter
         &item);
 
-    updateError(err, status);
-    if (!err) {
-        status = SecKeychainItemDelete(item);
+    if (status != errSecSuccess) {
         updateError(err, status);
+    } else {
+        status = SecKeychainItemDelete(item);
+        if (status != errSecSuccess) {
+            updateError(err, status);
+        }
+    }
+
+    if (item) {
```

Comment:
We didn't do this check before. It's required now if we don't fancy segfaults, but I honestly cannot figure out why :confused: 
The test case that segfaults (without this) is "delete when password is not found". In this case, we will `updateError`, because status is `errSecItemNotFound`. We won't go to the `else` branch.
Before, we'd `updateError` in any case, but also don't go into the conditional branch, because `err` will be set. What changed? Just "lucky" before?

Reply:
Thats indeed funny. However, checking for `nullptr` before `CFRelease` is definitely a good idea. If the tests still work, I'd put it off with "we were lucky before".